plugin/pkg/admission/gc/gc_admission.go

```diff
@@ -122,6 +123,18 @@ func (a *gcPermissionsEnforcement) Validate(ctx context.Context, attributes admi
 	// true. If so, only allows the change if the user has delete permission of
 	// the _OWNER_
 	newBlockingRefs := newBlockingOwnerDeletionRefs(attributes.GetObject(), attributes.GetOldObject())
+	if len(newBlockingRefs) == 0 {
+		return nil
+	}
+
+	// There can be a case where a restMapper tries to hit discovery endpoints and times out if the network is inaccessible.
+	// This can prevent creating the pod to run the network to be able to do discovery and it appears as a timeout, not a rejection.
```

Comment:
@deads2k how about changing the mapper to accept the `context` with the request deadline instead? The authorizer attributes set below seems to be covering only privileged users.

Reply:
> @deads2k how about changing the mapper to accept the `context` with the request deadline instead? The authorizer attributes set below seems to be covering only privileged users.

That wouldn't fix the overall problem we face since we want to have the request succeed.  As a separate effort, that improvement makes sense, but it wouldn't fix this problem.

Reply:
> That wouldn't fix the overall problem we face since we want to have the request succeed

how do we ensure that the next admission won't run into the discovery issue?

Reply:
> how do we ensure that the next admission won't run into the discovery issue?

Well, in this case because this is the only admission plugin that is a `WantsRESTMapper`.  I would say that if another admission plugin grows that requires it, we first discourage them and then consider repercussions for that admission plugin before making a change.